### ISSUE TITLE
[Question] How to get absolute pointer x,y

### ISSUE THREAD

**Issue by Arjun31415** [Smithay/input.rs#55]
I would like to get absolute pointer x and y coordinates, but that event is never triggered. Only `PointerEvent::Motion` is caught.

```rust
use input::event::pointer::PointerEvent as LibinputPointerEvent;
use input::{AsRaw, Libinput, LibinputInterface};
use nix::poll::{poll, PollFd, PollFlags};

struct Interface;

impl LibinputInterface for Interface {
    fn open_restricted(&mut self, path: &Path, flags: i32) -> Result<OwnedFd, i32> {
        OpenOptions::new()
            .custom_flags(flags)
            // Open as Read-Only, always
            .read(true)
            .write(false)
            .open(path)
            .map(|file| file.into())
            .map_err(|err| err.raw_os_error().unwrap())
    }
    fn close_restricted(&mut self, fd: OwnedFd) {
        drop(File::from(fd))
    }
}


fn main() {

    let mut input = Libinput::new_with_udev(Interface);
    input.udev_assign_seat("seat0").unwrap();
    let pollfd = PollFd::new(input.as_raw_fd(), PollFlags::POLLIN);
    while poll(&mut [pollfd], -1).is_ok() {
        input.dispatch().unwrap();
        for event in &mut input {
            if let input::event::Event::Pointer(LibinputPointerEvent::Motion(
                pointer_event,
            )) = &event
            {
                println!("({}, {})", pointer_event.dx(), pointer_event.dy());
            }
        }
    }
}

```

If I use  `MotionAbsolute` there is no event triggered. I have a mouse and trackpad but neither trigger that event. What could be the reason?.

**Comment by Drakulix** [comment]
Not every device type triggers this event. Devices that only provide relative movements (like mice or trackpads) will always create Motion events.

MotionAbsolute events are usually generated from devices with a fixed area, like touchscreens.

**Comment by Arjun31415** [comment]
I see. Then I have to go with a roundabout solution. I will make a layer using layershell and then listen for pointer entered event, store that and then destroy the layer, then use libinput event and calculate using dx,dy.

Thank you.

**Comment by FoxxMD** [comment]
@Arjun31415 can you elaborate on how you are getting absolute position with layershell? I'm trying to achieve the same result.

**Comment by Arjun31415** [comment]
When layer shell is created for the first time, it would give the (x,y) values of the mouse pointer, and then using (dx,dy) from libinput to calculate the absolute position (x+dx,y+dy). However there are many edge cases, say across edges where the pointer does not move beyond the edge libinput still emits (dx,dy) values. 

In the end I realized this was a security feature of wayland, not providing the absolute coordinates of mouse pointer to prevent tracking/snooping. However, the compositior I used - Hyprland, provided the absolute coordinates and I just used its dispatch commands to get the absolute position
